### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by richfitz**
Sharepoint remote for orderly

One point to consider; is the package name correct? I have a prototype dropbox implementation good to go too, which might be best put into the one package perhaps?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,45 @@
...
+      path: Shared Documents/orderly/testing
+```
+
+Where
+
+* `url` is the base url of your Office365/Sharepoint site, such as `myorg.sharepoint.com`
+* `site` is your site name on Sharepoint
+* `path` is the path within your site name where documents will be stored
+
+The configuration above lists two remotes, one "real" and one "testing", which we have found a useful pairing.  You might configure Sharepoint to allow anyone in your group to read from both, but only certain people to push to `real`.
```
Comment (r-ash) [inline README.md:37]: Leveraging sharepoints permissions is really cool

File: R/orderly.R [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+##'   real:
+##'     driver: orderly.sharepoint::orderly_remote_sharepoint
+##'     args:
+##'       url: https://example.sharepoint.com
+##'       site: mysite
+##'       path: Shared Documents/orderly/real
+##' }
+##'
+##' which would create a remote called \code{real}, using your group's
+##' Sharepoint hosted at \code{https://example.sharepoint.com}, on
```
Comment (r-ash) [inline R/orderly.R:17]: Maybe useful to add a note about authentication?

File: tests/testthat/test-orderly.R [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+test_that("list_reports calls folder$folders('archive')", {
+  folder <- list(folders = mockery::mock(list(name = c("a", "b", "c"))))
+  cl <- orderly_remote_sharepoint_$new(folder)
+  expect_equal(cl$list_reports(), c("a", "b", "c"))
+  mockery::expect_called(folder$folders, 1)
+  expect_equal(mockery::mock_args(folder$folders)[[1]], list("archive"))
+})
+
+
+test_that("list_versions calls folder$files('archive/:name')", {
```
Comment (r-ash) [inline tests/testthat/test-orderly.R:12]: Should this be `archive/name` not `:name`?

Comment (richfitz) [reply tests/testthat/test-orderly.R]: Ah, was trying to evoke a placeholder, have changed now to <name> which is more obvious

File: R/orderly.R (hunk 2 above)
Comment (richfitz) [reply R/orderly.R:17]: done, thanks

**Pull request merged**